### PR TITLE
fix: CalendarView maxdate scrolling

### LINKED ISSUE #17635: [CalendarView] Significant Gap Appears in Calendar When when reaching end of calendar
### Current behavior

After selecting a day at the end of the calendar and switching back to the year view, a large gap is present, disrupting the user experience and causing visual inconsistencies.

https://github.com/user-attachments/assets/250d6ff2-c146-4643-b912-27e8a7d80a7b

Screenshot 2024-07-22 at 11 39 45 AM

### Expected behavior

The calendar should smoothly transition between month and year views without any large gaps or layout issues.

### How to reproduce it (as minimally and precisely as possible)

Steps to Reproduce:

1. Open Gallery in any of the web browser
2. Navigate to Menu -> UI Components -> CalenderView
3. In the calendar reach to end of years
4. Select a day
5. Go back to years
6. Observe the huge blank gap.

### Workaround

_No response_

### Works on UWP/WinUI

None

### Environment

_No response_

### NuGet package version(s)

_No response_

### Affected platforms

WebAssembly

### IDE

_No response_

### IDE version

_No response_

### Relevant plugins

_No response_

### Anything else we need to know?

_No response_

### PR DESCRIPTION
**Pull request by rajamatt**
GitHub Issue (If applicable): closes #17635

## PR Type

What kind of change does this PR introduce?

- Bugfix

## What is the current behavior?

Scrolling to the MaxDate of the CalendarView and then switching views causes items to disappear.

https://github.com/user-attachments/assets/95c27b13-a069-42ea-8855-18dde8226c1e

## What is the new behavior?

https://github.com/user-attachments/assets/bc551a96-f3c2-48e5-8a71-61c1441e4630

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [x] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [x] Contains **NO** breaking changes
- [x] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [x] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: src/Uno.UI/UI/Xaml/Controls/CalendarView/Primitives/CalendarPanel.ModernCollectionBasePanel.cs [hunk 1]
```diff
@@ -432,7 +432,15 @@ internal void ScrollItemIntoView(int index, ScrollIntoViewAlignment alignment, d
 				// then it will request GetContainerFromIndex and tries to focus it.
 				// So here we prepare the _effectiveViewport (which will most probably be re-updated by the ChangeView below),
 				// and then force a base_Measure()
-				_effectiveViewport.Y += newOffset - currentOffset;
+
+				if (newOffset >= sv.ScrollableHeight)
```
Comment (MartinZikmund) [inline src/Uno.UI/UI/Xaml/Controls/CalendarView/Primitives/CalendarPanel.ModernCollectionBasePanel.cs:436]: If this is not coming from WinUI sources, then it should be wrapped in `#if HAS_UNO` with a `//` explanation of the issue

**Review by Xiaoy312 — APPROVED** [review]

File: src/Uno.UI/UI/Xaml/Controls/CalendarView/Primitives/CalendarPanel.ModernCollectionBasePanel.cs (hunk 1 above)
Comment (rajamatt) [reply src/Uno.UI/UI/Xaml/Controls/CalendarView/Primitives/CalendarPanel.ModernCollectionBasePanel.cs:439]: done!

**Pull request merged**